src/widgets/scrollbar.rs

```diff
@@ -1128,4 +1131,341 @@ mod tests {
+        #[case] expected: &str,
+        #[case] position: usize,
+        #[case] content_length: usize,
+        #[case] assertion_message: &str,
+    ) {
+        let size = expected.width() as u16;
+        let mut buffer = Buffer::empty(Rect::new(0, 0, size, 1));
+        let mut state = ScrollbarState::default()
+            .position(position)
+            .content_length(content_length);
+        Scrollbar::default()
+            .orientation(ScrollbarOrientation::HorizontalBottom)
+            .begin_symbol(None)
+            .end_symbol(None)
+            .render(buffer.area, &mut buffer, &mut state);
+        assert_eq!(
+            buffer,
+            Buffer::with_lines(vec![expected]),
+            "{}",
+            assertion_message
+        );
+    }
+
+    #[rstest]
+    #[case("<######-->", 0, 10, "position_0")]
+    #[case("<######-->", 1, 10, "position_1")]
+    #[case("<######-->", 2, 10, "position_2")]
+    #[case("<-######->", 3, 10, "position_3")]
+    #[case("<-######->", 4, 10, "position_4")]
+    #[case("<-######->", 5, 10, "position_5")]
+    #[case("<-######->", 6, 10, "position_6")]
+    #[case("<-######->", 7, 10, "position_7")]
+    #[case("<--######>", 8, 10, "position_8")]
+    #[case("<--######>", 9, 10, "position_9")]
+    #[case("<--######>", 10, 10, "position_one_out_of_bounds")]
+    #[case("<--######>", 15, 10, "position_few_out_of_bounds")]
+    #[case("<--######>", 500, 10, "position_very_many_out_of_bounds")]
+    fn render_scrollbar_with_symbols(
+        #[case] expected: &str,
+        #[case] position: usize,
+        #[case] content_length: usize,
+        #[case] assertion_message: &str,
+    ) {
+        let size = expected.width() as u16;
+        let mut buffer = Buffer::empty(Rect::new(0, 0, size, 1));
+        let mut state = ScrollbarState::default()
+            .position(position)
+            .content_length(content_length);
+        Scrollbar::default()
+            .orientation(ScrollbarOrientation::HorizontalTop)
+            .begin_symbol(Some("<"))
+            .end_symbol(Some(">"))
+            .track_symbol(Some("-"))
+            .thumb_symbol("#")
+            .render(buffer.area, &mut buffer, &mut state);
+        assert_eq!(
+            buffer,
+            Buffer::with_lines(vec![expected]),
+            "{}",
+            assertion_message,
+        );
+    }
+
+    #[rstest]
+    #[case("██████████", 0, 10, "position_0")]
+    #[case("██████████", 1, 10, "position_1")]
+    #[case("██████████", 2, 10, "position_2")]
+    #[case("██████████", 3, 10, "position_3")]
+    #[case("██████████", 4, 10, "position_4")]
+    #[case("██████████", 5, 10, "position_5")]
+    #[case("██████████", 6, 10, "position_6")]
+    #[case("██████████", 7, 10, "position_7")]
+    #[case("██████████", 8, 10, "position_8")]
+    #[case("██████████", 9, 10, "position_9")]
+    #[case("██████████", 100, 10, "position_out_of_bounds")]
+    fn render_scrollbar_twoline_horizontal(
```

Comment:
So here you're testing when there is a top and bottom bar right? Have you noticed weird behavior related to that?

Reply:
+1 this is confusing